scripts/ci/pr-bot/processPrUpdate.ts

```diff
@@ -0,0 +1,198 @@
+) {
+  if (payload.review.state != "approved") {
+    return;
+  }
+
+  const pullNumber = payload.issue?.number || payload.pull_request?.number;
+  if (!(await areReviewersAssigned(pullNumber, stateClient))) {
+    return;
+  }
+
+  let prState = await stateClient.getPrState(pullNumber);
+  // TODO(damccorm) - also check if the author is a committer, if they are don't auto-assign a committer
+  if (await prState.isAnyAssignedReviewerCommitter()) {
+    return;
+  }
+
+  let labelOfReviewer = prState.getLabelForReviewer(payload.sender.login);
+  if (labelOfReviewer) {
+    let reviewersState = await stateClient.getReviewersForLabelState(
+      labelOfReviewer
+    );
+    let availableReviewers =
+      reviewerConfig.getReviewersForLabel(labelOfReviewer);
+    let chosenCommitter = await reviewersState.assignNextCommitter(
+      availableReviewers
+    );
+    prState.reviewersAssignedForLabels[labelOfReviewer] = chosenCommitter;
+
+    // Set next action to committer
+    await addPrComment(
+      pullNumber,
+      commentStrings.assignCommitter(chosenCommitter)
+    );
+    const existingLabels =
+      payload.issue?.labels || payload.pull_request?.labels;
+    await nextActionReviewers(pullNumber, existingLabels);
+    prState.nextAction = "Reviewers";
+
+    // Persist state
+    await stateClient.writePrState(pullNumber, prState);
+    await stateClient.writeReviewersForLabelState(
+      labelOfReviewer,
+      reviewersState
+    );
+  }
+}
+
+// On pr push or author comment, we should put the attention set back on the reviewers
+async function setNextActionReviewers(
+  payload,
+  stateClient: typeof PersistentState
+) {
+  const pullNumber = payload.issue?.number || payload.pull_request?.number;
+  if (!(await areReviewersAssigned(pullNumber, stateClient))) {
+    console.log("No reviewers assigned, dont need to manipulate attention set");
+    return;
+  }
+  const existingLabels = payload.issue?.labels || payload.pull_request?.labels;
+  await nextActionReviewers(pullNumber, existingLabels);
+  let prState = await stateClient.getPrState(pullNumber);
+  prState.nextAction = "Reviewers";
+  await stateClient.writePrState(pullNumber, prState);
+}
+
+async function processPrUpdate() {
+  const reviewerConfig = new ReviewerConfig(PATH_TO_CONFIG_FILE);
+  const context = github.context;
+  console.log("Event context:");
+  console.log(context);
+  const payload = context.payload;
+
+  // TODO(damccorm) - remove this when we roll out to more than go
+  const existingLabels = payload.issue?.labels || payload.pull_request?.labels;
+  if (!existingLabels.find((label) => label.name.toLowerCase() == "go")) {
+    console.log("Does not contain the go label - skipping");
+    return;
+  }
+
+  if (!payload.issue?.pull_request && !payload.pull_request) {
+    console.log("Issue, not pull request - returning");
+    return;
+  }
+  const pullNumber = payload.issue?.number || payload.pull_request?.number;
```

Comment:
The pullNumber access pattern is used 3 times in this PR. Please add a helper method.